### DISCUSSION TITLE
Deleting language files

### DISCUSSION THREAD

**Discussion by 737simpilot** [CIDRAM/CIDRAM#418] [General]
Can I just delete the extra language files I don't need or is there another file in play here I have to mess with as well? I wanted to post this at Gitter but I see they went full on crap mode with Element and not only does it take longer to load, but its blinding white and like everything these days that is considered an "upgrade", it's a massive downgrade. So I won't be using it from here on out.

**Comment by Maikuolan** [comment]
> Can I just delete the extra language files I don't need

Yep, you can. :-)

>  or is there another file in play here I have to mess with as well?

Just removing them outright won't break anything (as long as you don't delete those for the currently configured language, of course). Ideally though, to prevent the updater from just reinstalling all those files again the next time you hit "update all" at the updates page, you'd want to have them removed from the components metadata file, too (which, "uninstalling" them from the updates page will do automatically, though, may be a little annoying to do, due to how many there are). Otherwise, if not removed from the components metadata file, CIDRAM will think they're still installed, and when an update for them becomes available, fetch the updates, and recreate them all.

Due to how many different components now exist, and how many of them are related to language files, and similarly, other things which not necessarily everyone would want or need at their installations, and due to the time required to click uninstall for each of those components, I'd been planning on implementing some kind of macros-like feature into the updates page at some point hopefully soon (e.g., 29 currently supported languages in v3, minus one seeing as one would need at least one language installed, e.g., English, in order to be able to read anything at all, so 28.. multiplied by two as language files are now split into two components, one for the core and one for the front-end plus CLI and other related stuff which uses the front-end.. is 56.. so.. possibly, deleting it all, one component at a time, would mean ~56 times actioning at the updates page to get it all done).

Had been thinking, if there was a way to perform multiple, conditional grouped actions at the updates page (e.g., "delete all L10N components except for those I'm currently using"), it could help users to save a bit of time and frustration, I think.

> I wanted to post this at Gitter but I see they went full on crap mode with Element and not only does it take longer to load, but its blinding white and like everything these days that is considered an "upgrade", it's a massive downgrade. So I won't be using it from here on out.

Yeah.. Understandable. Not too keen on the recent changes myself, either. Well.. We'd been thinking of slowly phasing out Gitter eventually in favour of just using GitHub Discussions instead anyway, so.. I guess, this just speeds that up a bit.

**Comment by 737simpilot** [comment]
👍

**Comment by Maikuolan** [comment]
Support for macros has now been added to the updates page, and it's possible now to install/uninstall/update *all* L10N (separated by core versus front-end, and excluding that for whichever language is currently set by the configuration) at the same time, with a single click. :-)

**Comment by 737simpilot** [comment]
👍
